### DISCUSSION TITLE
Spatie Laravel Permission- Revoking access after role changed

### DISCUSSION THREAD

**Discussion by alexraynes** [spatie/laravel-permission#1919] [Q&A]
Greetings, please provide your support. 
Spatie Laravel Permission. I have deployed it in my laravel application but experienced this issue. In my application users have roles and roles have permissions which is correct, but when I assign a user to a super admin role and later change that user role/type to editor it still displays the super admin permission access to that user.  How can the access be revoked to the new role permissions access right after changing the role?
am using the @can in the blade

**Comment by erikn69** [comment]
[assigning-roles](https://spatie.be/docs/laravel-permission/v5/basic-usage/role-permissions#assigning-roles)

>A role can be removed from a user:
>
>$user->removeRole('writer');

**Reply by alexraynes** [reply]
Hi Erikn, thanks for your reply. 
permissions are successfully  revoking from role, but the issue has to do with.
Example: I have role 
Admin: permission ( manage user, Create ticket, Manage System setting)
Editor: permission ( edit user, edit ticket, and Edit System setting)  
Viewer: permission (View user, View ticket, and View System setting  )
The role are save in my db as user type.
When I go to manage user from my UI/blade to switch the user role from Viewer to Admin it successfully does that and when the user login he/she has all the permission of the Admin, but my issue is when I decide to switch that same user role from Admin to Viewer the the user type in my db successfully changes from Admin to Viewer, but when that user login they still have the Admin privilege/permission .   Please help if you have an idea of what happening.

**Reply by erikn69** [reply]
>The role are save in my db as user type.

I didn't understand that

------
Looks like an issue from your implementation, i do the same and works as expected, also the [tests](https://github.com/spatie/laravel-permission/tree/main/tests) works
If you want help, you have to create a minimal Laravel app that demonstrates the issue, or maybe create a failing tests for look at, [example](https://github.com/Onekone/spatie-issue-example/blob/main/tests/Feature/IssueTest.php)
